aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -63,6 +63,29 @@ static inline std::tuple<Tensor, Tensor> _lu_det_P_diag_U(const Tensor& self) {
   return std::tuple<Tensor, Tensor>(num_exchanges.mul_(-2).add_(1), u_diagonal);
 }
 
+// Given a pivoted LU factorization A = P L U,
+// det(A) = det(P) * det(L) * det(U).
+// Since det(P) = +- 1 (even or odd permutation), and diag(L) = I, we get that
+// det(A) = ([is P odd] * -2 + 1) * prod(diag(U))
+std::tuple<Tensor, Tensor, Tensor, Tensor> _det_lu_based_helper(const Tensor& self) {
```

Comment:
`_lu_det_P_diag_U` and `_det_lu_based_helper` are very similar. The only differences are:
* The extra `lu_unpack` call
* The extra return values
* The extra multiply in the line `auto det = p_det * at::prod(...`

Would it be beneficial to reduce duplication? For instance, could change `_lu_det_P_diag_U` to optionally call `lu_unpack` (toggled with a bool argument), and then `_det_lu_based_helper` could use `_lu_det_P_diag_U`.

Reply:
Then we would need to backprop thought prod. It is possible to avoid that.